Signal/src/util/ThreadUtil.m

```diff
@@ -230,6 +230,30 @@ + (void)createUnreadMessagesIndicatorIfNecessary:(TSThread *)thread storageManag
     }];
 }
 
++ (void)clearUnreadMessagesIndicator:(TSThread *)thread storageManager:(TSStorageManager *)storageManager
+{
+    OWSAssert(thread);
+    OWSAssert(storageManager);
+
+    [storageManager.dbConnection readWriteWithBlock:^(YapDatabaseReadWriteTransaction *transaction) {
+
+        NSMutableArray *indicators = [NSMutableArray new];
+        [[transaction ext:TSMessageDatabaseViewExtensionName]
+            enumerateRowsInGroup:thread.uniqueId
+                      usingBlock:^(
+                          NSString *collection, NSString *key, id object, id metadata, NSUInteger index, BOOL *stop) {
+
+                          if ([object isKindOfClass:[TSUnreadIndicatorInteraction class]]) {
```

Comment:
Let's do it in the next PR to avoid rebase hell.  It'll be painful though because any device infected with "unread indicators" will crash when trying to deserialize "unseen indicators".